### ISSUE TITLE
t-vim doesn't appear to set vimrc correctly

### ISSUE THREAD

**Issue by Semptum** [adityam/filter#45]
I was trying to make Julia syntax highlighting work. When I started looking in the logs, I saw that vim was called with vim -u NONE so I set 
```
\definevimtyping[julia][
    syntax=julia,
    vimrc=~/.vimrc,
    escape=on
]
```

However after that the logs showed something weird:
```
vim -u sujet-temp-vimrc-~/.vimrc.tmp -X -i NONE --noplugin --clean -es -n -c "syntax manual" -c "set syntax=julia" -c "set tabstop=4" -c "let contextstartline=1| let contextstopline=0| let strip=1" -c "let escapecomments=1" -c "let highlight=[]"  -c "source /usr/share/texmf-dist/tex/context/third/vim/2context.vim" -c "qa" sujet-temp-julia-0.tmp sujet-temp-julia-0.vimout
```
 To avoid confusion, my filename is "sujet.tex".

I'm still looking trying to understand the source code but this vimrc is just weird and without setting the correct vimrc the syntax highlighting doesn't work even in interactive mode. Is it normal behaviour?

Thanks for the work on this module btw, it's pretty great for other languages.

Best regards,

**Comment by Semptum** [comment]
So, I can generate a correct .vimout file with the right syntactic information but I have to do the following:
- replace --clean with -N
- provide a correct .vimrc file in the -u option
So, to summarize, the command went from 

```
vim -u NONE -X -i NONE --noplugin --clean -es -n -c "syntax manual" -c "set syntax=julia" -c "set tabstop=4" -c "let contextstartline=1| let contextstopline=0| let strip=1" -c "let escapecomments=1" -c "let highlight=[]"  -c "source /usr/share/texmf-dist/tex/context/third/vim/2context.vim" -c "qa" sujet-temp-julia-0.tmp sujet-temp-julia-0.vimout
```
to
```
vim -u ~/.vimrc -X -i NONE --noplugin -N -es -n -c "syntax manual" -c "set syntax=julia" -c "set tabstop=4" -c "let contextstartline=1| let contextstopline=0| let strip=1" -c "let escapecomments=1" -c "let highlight=[]"  -c "source /usr/share/texmf-dist/tex/context/third/vim/2context.vim" -c "qa" sujet-temp-julia-0.tmp sujet-temp-julia-0.vimout
```

And the output went from
```
\SYNBOL{}iter=Fibonacci(n)\SYNEOL{}
\SYNBOL{}for i in iter\SYNEOL{}
\SYNBOL{}    #body\SYNEOL{}
\SYNBOL{}end\SYNEOL{}
```

to
```
\SYNBOL{}iter\SYN[Operator]{=}\SYN[juliaNone]{Fibonacci}\SYN[juliaNone]{(}\SYN[juliaParBlock]{n}\SYN[juliaNone]{)}\SYNEOL{}
\SYNBOL{}\SYN[Repeat]{for}\SYN[juliaForBlock]{ i }\SYN[Keyword]{in}\SYN[juliaForBlock]{ iter}\SYNEOL{}
\SYNBOL{}\SYN[juliaForBlock]{    }\SYN[Comment]{#}\SYN[Comment]{body}\SYNEOL{}
\SYNBOL{}\SYN[Repeat]{end}\SYNEOL{}
```

It could be that I misunderstood something basic and there is a reason why it uses --clean and prepends something to the vimrc filename, I'm not fluent in ConTexT yet.

**Comment by adityam** [comment]
Are you using the latest version of the `t-vim` module? There was a bug which was fixed recently. See e2a606d9964ecc5abdabf885142b5a73fdfccecb

**Comment by adityam** [comment]
The vimrc setting is not a bug. See the section on [tuning color schemes](https://github.com/adityam/filter/blob/dev/vim-README.md#tuning-color-schemes) to see how it is supposed to be used. The t-vim module has no mechanism for loading a system wide vimrc (which is typically meant for interactive usage and has the potential slow things down)

**Comment by Semptum** [comment]
> Are you using the latest version of the `t-vim` module? There was a bug which was fixed recently. See [e2a606d](https://github.com/adityam/filter/commit/e2a606d9964ecc5abdabf885142b5a73fdfccecb)
> 
> Is there a specific reason why you want to set the non-compatible mode?

I am now, yes.

I needed to set the non-compatible mode to load the julia plugin, otherwise the syntax highlighting doesn't work. But I was reading the man for the vim command options too much and forgot it was also possible in the vimrc. I have implemented a vimrc that works thanks to your comment.

Apologies for the issue, a forum might have been more suited. 
Since the latest version works without issues to produce a syntactically annotated file and it was user error that caused the problems, I'm closing the issue.
